caffe2/core/export_caffe2_op_to_c10.h

```diff
@@ -29,7 +29,7 @@ inline c10::List<at::Tensor> _call_caffe2_op(
     std::vector<c10::IValue>&& inputs,
     c10::List<at::Tensor>&& outputs) {
   Caffe2Operator op(schema, std::move(inputs), std::move(outputs));
-  op.Run();
+  op.Run(-1);
```

Comment:
@ezyang, I believe it is the line above this (constructor of `caffe2::Operator` [[operator.h#L814](https://github.com/pytorch/pytorch/blob/7824d8277aebc20ac8c21c5f4b52a29182c24816/caffe2/core/operator.h#L814)] that does the initial switching to device, and thus the additional stream is already created and switched to here.

Reply:
Good catch.